libyul/optimiser/ExpressionInliner.h

```diff
@@ -38,7 +38,10 @@ struct Dialect;
  *  - have a body like r := <functional expression>
  *  - neither reference themselves nor r in the right hand side
  *
- * Furthermore, the arguments of the function call cannot have any side-effects.
+ * Furthermore, for all parameters, all of the following need to be true
+ *  - the argument is movable
+ *  - the parameter is either referenced less than twice in the function body, or the argument is rather cheap
```

Comment:
`the argument` instead of `the parameter`? Also, could you please line break after `in the function body` for readability?

Reply:
can the function argument be never referenced in function body? In other words, could the comment be reworded as `the argument is referenced exactly once in the function body, or ...`?

Reply:
> `the argument` instead of `the parameter`? Also, could you please line break after `in the function body` for readability?

Actually the use of "parameter" and "argument" at the same time was intended because I thought they have different meanings here. I use "parameter" to refer to variables in the the function definition's header, and use "argument" to refer to the expressions that are going to be passed to the function. But obviously I'm not a native English user so I'm not sure if this mixed usage of "parameter" and "argument" is confusing or sounds strange/unnatural.

> can the function argument be never referenced in function body?

I think it can?

Reply:
Oh okay, that sounds good to me. Could the `argument is rather cheap` be clarified to something like the `cost of the argument is at most one`?

Reply:
Yes I can change that. (`argument is rather cheap` was actually copy-pasted from Rematerializer's comment)